ios/Classes/BaseCarSceneDelegate.swift

```diff
@@ -0,0 +1,146 @@
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import CarPlay
+import Foundation
+import GoogleMaps
+
+open class BaseCarSceneDelegate: UIResponder, CPTemplateApplicationSceneDelegate,
+  CPMapTemplateDelegate {
+  private var interfaceController: CPInterfaceController?
+  private var carWindow: CPWindow?
+  private var mapTemplate: CPMapTemplate?
+  private var navView: GoogleMapsNavigationView?
+  private var navViewController: UIViewController?
+  private var templateApplicationScene: CPTemplateApplicationScene?
+  private var autoViewEventApi: AutoViewEventApi?
+  private var viewRegistry: GoogleMapsNavigationViewRegistry?
+
+  public func templateApplicationScene(_ templateApplicationScene: CPTemplateApplicationScene,
+                                       didConnect interfaceController: CPInterfaceController,
+                                       to window: CPWindow) {
+    self.interfaceController = interfaceController
+    carWindow = window
+    mapTemplate = getTemplate()
+    self.templateApplicationScene = templateApplicationScene
+    mapTemplate?.mapDelegate = self
+    createVC()
+  }
+
+  open func getTemplate() -> CPMapTemplate {
+    let template = CPMapTemplate()
+    template.showPanningInterface(animated: true)
+    return template
+  }
+
+  open func templateApplicationScene(_ templateApplicationScene: CPTemplateApplicationScene,
+                                     didDisconnect interfaceController: CPInterfaceController,
+                                     from window: CPWindow) {
+    self.interfaceController = nil
+    carWindow?.rootViewController = nil
+    carWindow = nil
+    mapTemplate = nil
+    navView?.unregisterView()
+    navView = nil
+    navViewController = nil
+    self.templateApplicationScene = nil
+  }
+
+  open func sceneDidBecomeActive(_ scene: UIScene) {}
+
+  func createVC() {
+    guard
+      let templateApplicationScene,
+      navView == nil
+    else { return }
+
+    GoogleMapsNavigationPlugin
+      .pluginInitializedCallback = { [weak self] viewRegistry, autoViewEventApi, imageRegistry in
+        guard let self else { return }
+        self.viewRegistry = viewRegistry
+        self.autoViewEventApi = autoViewEventApi
+        self.navView = GoogleMapsNavigationView(
+          frame: templateApplicationScene.carWindow.screen.bounds,
+          isNavigationView: true,
+          viewRegistry: viewRegistry,
+          navigationUIEnabledPreference: NavigationUIEnabledPreference.automatic,
+          mapConfiguration: MapConfiguration(
+            cameraPosition: GMSCameraPosition(latitude: 51.51, longitude: 0.12, zoom: 5),
```

Comment:
What is this?
why we have hardcoded location here?

Reply:
Done.